cmd/operator-sdk/olmcatalog/cmd.go

```diff
@@ -15,16 +15,70 @@
 package olmcatalog
 
 import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold/input"
+	catalog "github.com/operator-framework/operator-sdk/internal/pkg/scaffold/olm-catalog"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
+var (
+	writeCSVConfigOpt = "write-csv-config"
+
+	writeCSVConfigPath string
+)
+
 func NewCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "olm-catalog <olm-catalog-command>",
+		Use:   "olm-catalog [flags] <olm-catalog-command>",
 		Short: "Invokes a olm-catalog command",
 		Long: `The operator-sdk olm-catalog command invokes a command to perform
 Catalog related actions.`,
+		RunE: olmCatalogFunc,
 	}
 	cmd.AddCommand(newGenCSVCmd())
+
+	i, err := (&catalog.CSVConfig{}).GetInput()
+	if err != nil {
+		log.Fatalf("Error retrieving CSV config path: %v", err)
+	}
+	cmd.PersistentFlags().StringVar(&writeCSVConfigPath, writeCSVConfigOpt, "",
+		"Write a default CSV config file. A default path is used if no value is provided."+
+			" Set --write-csv-config=<path> to supply a non-default path")
+	cmd.Flag(writeCSVConfigOpt).NoOptDefVal = i.Path
+
 	return cmd
 }
+
+func olmCatalogFunc(cmd *cobra.Command, args []string) error {
+	if cmd.Flags().Changed(writeCSVConfigOpt) {
+		absProjectPath := projutil.MustGetwd()
+		cfg := &input.Config{
+			AbsProjectPath: absProjectPath,
+			ProjectName:    filepath.Base(absProjectPath),
+		}
+		if projutil.IsOperatorGo() {
+			cfg.Repo = projutil.CheckAndGetProjectGoPkg()
+		}
+		s := &scaffold.Scaffold{}
+		if err := writeConfig(s, cfg); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
Is it creating a new input.Config? Could not it be a NewConfig in its struc, or at least has a more meaningful name?